src/LinuxTracing/PerfEvent.h

```diff
@@ -249,6 +249,24 @@ struct DmaFenceSignaledPerfEventData {
 };
 using DmaFenceSignaledPerfEvent = TypedPerfEvent<DmaFenceSignaledPerfEventData>;
 
+struct PerfRecordSample {
```

Comment:
I think if we do a generic approach here, we might want to consider all possible PERF_RECORD_SAMPLE combinations. But I'd also like to have @beckerhe's opinion on this.

Reply:
I'd agree if this was a public API. In a public API we need to convey way more clearly which fields are present and which are not. But this doesn't need to be public, right? Why is the type in the header? Couldn't you move it into the cpp file?

(Nevertheless, the struct should have a comment on top which explains that not all fields are always present and the user needs to take care theirselves.)

Reply:
I have made it more generic now. It should be more similar to perf record sample from perf event.

Reply:
Thanks. Looks much better now. However one nit comment:
As per style guide, the struct needs a brief comment, also about its usage. Most importantly you need to explain that this is reflecting the perf_event_open "struct" from the man page and that, if we need to add fields, we need to uncomment them and handle them in the consumer respectively.